dropshot/tests/fail/bad_endpoint3.stderr

```diff
@@ -1,32 +1,3 @@
-error[E0277]: the trait bound `String: Extractor` is not satisfied
-  --> tests/fail/bad_endpoint3.rs:17:12
-   |
-17 |     param: String,
-   |            ^^^^^^ the trait `Extractor` is not implemented for `String`
```

Comment:
Totally.  Since you're on board with this general idea, the next step is to figure out how to ensure good error messages when (1) an argument is not an extractor at all, or (2) you attempt to use two exclusive extractors.  I think (1) is pretty easy because I made `SharedExtractor` impl `ExclusiveExtractor`, so we can change the existing check to check `ExclusiveExtractor`.  I have not yet figured out a good way to do (2) but Rain suggested an approach I'm going to look at.

Reply:
Done.  The basic approach is to verify that:

- the first argument is a `RequestContext` (like we did before)
- the last argument is an `ExclusiveExtractor` (slight variation on what we did before)
- all other arguments are `SharedExtractor`

This has the side effect of verifying that you don't have two exclusive extractors because they don't impl SharedExtractor.

The naming of the tests is a little weird.  I kept the existing dense integer naming but it's a little confusing that test 19 is a slight variant of test 3.  (That is, the behavior tested by the old bad_endpoint3.rs is now tested by a combination of bad_endpoint3.rs and bad_endpoint19.rs.)  The alternatives seemed like separating into 3a and 3b or renaming everything to be more semantic. 🤷